docs/StakingPool.md

```diff
@@ -0,0 +1,289 @@
+
+
+```solidity
+function withdraw(uint tokenId, bool withdrawStake, bool withdrawRewards, uint[] memory trancheIds) external;
+```
+
+| Parameter         | Description                                       |
+|-------------------|---------------------------------------------------|
+| `tokenId`         | The ID of the staking NFT token.                  |
+| `withdrawStake`   | Whether to withdraw the stake.                    |
+| `withdrawRewards` | Whether to withdraw the rewards.                  |
+| `trancheIds`      | The IDs of the expired tranches to withdraw from. |
+
+### `extendDeposit`
+Extends the duration of a deposit by moving it from an active tranche to a future tranche.
+
+
+```solidity
+function extendDeposit(uint tokenId, uint initialTrancheId, uint targetTrancheId, uint topUpAmount) external;
+```
+
+| Parameter          | Description                          |
+|--------------------|--------------------------------------|
+| `tokenId`          | The ID of the staking NFT token.     |
+| `initialTrancheId` | The ID of the initial tranche.       |
+| `targetTrancheId`  | The ID of the target tranche.        |
+| `topUpAmount`      | The amount to top up the deposit by. |
+
+
+## View Functions
+
+
+### `getDeposit`
+Returns deposit data by token id and trancheId.
+
+```solidity
+function getDeposit(uint tokenId, uint trancheId) external override view returns (
+  uint lastAccNxmPerRewardShare,
+  uint pendingRewards,
+  uint stakeShares,
+  uint rewardsShares
+);
+```
+
+| Parameter        | Description            |
+|------------------|------------------------|
+| `tokenId`        | The ID of the product. |
+| `trancheId`      | The ID of the tranche. |
+
+### `getTranche`
+Returns tranche data by tranche id.
+
+```solidity
+function getTranche(uint trancheId) external override view returns (
+  uint stakeShares,
+  uint rewardsShares
+);
+```
+
+| Parameter        | Description            |
+|------------------|------------------------|
+| `trancheId`      | The ID of the tranche. |
+
+### `getExpiredTranche`
+Returns expired tranche data by tranche id.
+
+```solidity
+function getExpiredTranche(uint trancheId) external override view returns (
+  uint accNxmPerRewardShareAtExpiry,
+  uint stakeAmountAtExpiry,
+  uint stakeSharesSupplyAtExpiry
+);
+```
+
+| Parameter        | Description            |
+|------------------|------------------------|
+| `trancheId`      | The ID of the tranche. |
+
+### `getActiveAllocations`
+Returns an array of the active allocations for a product per tranche.
+
+```solidity
+function getActiveAllocations(uint productId)  public view returns (uint[] memory trancheAllocations);
+```
+
+| Parameter          | Description                          |
+|--------------------|--------------------------------------|
+| `productId`        | The ID of the product.               |
+
+### `getActiveTrancheCapacities`
+Returns an array of the active tranche capacities and total capacity for a product.
```

Comment:
Returns an array with the total capacities for the currently active tranches for a product.